docs/analyzer/decision_process.md

```diff
@@ -1,3 +1,80 @@
+```
+
+The format of the traces is: `[Date Time][decision_process][Log Level][Unique Correlation ID][Trace Message]`
+
+## Writing custom decision process for a recognizer
+
+When setting up the AnalyzerEngine with `enable_trace_pii=True`, the traces are written automatically. It means that when you add a new recognizer, a generic decision process trace will be written.
+
+However, it's possible to append custom information to the decision process if you wish to.
+
+For example, the [spacy_recognizer.py](https://github.com/microsoft/presidio/blob/master/presidio-analyzer/presidio_analyzer/predefined_recognizers/spacy_recognizer.py) implements a custom trace as follows:
+
+```python
+SPACY_DEFAULT_EXPLANATION = "Identified as {} by Spacy's Named Entity Recognition"
+
+def build_spacy_explanation(recognizer_name, original_score, entity):
+    explanation = AnalysisExplanation(
+        recognizer=recognizer_name,
+        original_score=original_score,
+        textual_explanation=SPACY_DEFAULT_EXPLANATION.format(entity))
+    return explanation
+```
+
+The `textual_explanation` field in `AnalysisExplanation` class allows you to add your own custom text into the final trace which will be written.
+
+## Enabling/Disabling Traces
+
+Decision-process traces are controlled by the `enable_trace_pii` parameter.
+
+```python
+from presidio_analyzer import AnalyzerEngine
+
+# Set up the engine, loads the NLP module (spaCy model by default) and other PII recognizers
+analyzer = AnalyzerEngine(enable_trace_pii=True)
+
+# Call analyzer to get results
+results = analyzer.analyze(text="My phone number is 212-555-5555",
+                           entities=["PHONE_NUMBER"],
+                           language='en')
+print(results)
+```
```

Comment:
Is there any impact for traces being enabled? in terms of runtime?
Is it ok to be enabled in production?
Is the memory impacted in any way?

Reply:
Shouldn't be any impact, same as writing any other log.

Reply:
Maybe we should state that :)

Reply:
Added